src/main/webapp/provider/appointmentprovideradminday.jsp

```diff
@@ -386,9 +386,26 @@
     int curMonth = (cal.get(Calendar.MONTH) + 1);
     int curDay = cal.get(Calendar.DAY_OF_MONTH);
 
-    int year = Integer.parseInt(request.getParameter("year"));
-    int month = Integer.parseInt(request.getParameter("month"));
-    int day = Integer.parseInt(request.getParameter("day"));
+    // Retrieve the 'year' parameter from the request and parse it into an integer
+    // If the parameter is null or empty, default to the current year
+    int year = Optional.ofNullable(request.getParameter("year"))
+                   .filter(param -> !param.isEmpty())
+                   .map(Integer::parseInt)
```

Comment:
@kateyang1998 - please respond to all of the Sourcery comments, even if it is just explaining why you aren't going to do it.  I think this would make sense to implement? I will let it merge to fix the issue, but Sourcery, can you make an issue ticket to track fixing this.